### DISCUSSION TITLE
Livewire mixing models of nested blade components

### DISCUSSION THREAD

**Discussion by eddysantos** [livewire/livewire#3794] [1. Help] [answered]
Hello everyone.

So, not sure if I'm doing massively wrong, I can't find a solution anywhere.

I have these two blade components:

modal.blade.php

```blade
@props(['width'])

@php
$width = $width ?? "w-11/12";
@endphp
<div x-data="{show: @entangle($attributes->wire('model'))}">
    <button
        type="button"
        class="text-lime-800 font-bold block w-max ml-auto p-2 rounded hover:bg-lime-600 hover:text-lime-200 hover:shadow"
        @click="show = true"
    >
        {{ $button_text ?? 'Open Modal' }}
    </button>
    <div
        x-show="show"
        x-transition
        {{ $attributes->merge(['class' => 'w-screen h-screen bg-black bg-opacity-30 fixed inset-0 pt-12']) }}
    >
        <div class="bg-white shadow mx-auto rounded {{ $width }} flex flex-col">
            <div class="p-3 border-b border-gray-200 flex-shrink flex justify-between bg-gray-50 rounded-t">
                <h5 class="text-2xl">{{ $title }}</h5>
                <button class="text-base transform hover:text-red-500 hover:scale-125" @click="show = false">
                    <svg xmlns="http://www.w3.org/2000/svg" class="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
                        <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M6 18L18 6M6 6l12 12" />
                    </svg>
                </button>
            </div>
            <div class="p-3">{{ $slot }}</div>
            @if (isset($footer))
                <div class="p-3">{{ $footer }}</div>
            @endif
        </div>
    </div>
</div>
```
and this one

text-input.blade.php

```blade
<div {{ $attributes->merge(['class'=> 'flex flex-col']) }}>
    <label for="{{ $id }}">{{ $label }}</label>
    <input
        type="text"
        class="py-1 px-2"
        id="{{ $id }}"
        placeholder="{{ $placeholder }}"
        {{ $attributes->wire('model') }}
    >
</div>
```

The problem is, whenever I call the text-input component inside the modal component, the 'show_model' property (which controls wether the model should be shown) gets updated whenever I type something in the input. Am I doing something wrong here? Or is this something that one cannot do?

Here's the view where I'm trying to use the components:

```blade
<div>
    <x-modal width="w-2/3" wire:model="show_modal">
        <x-slot name="button_text">@fa('plus') Nueva Villa</x-slot>
        <x-slot name="title" >Agregar Villa</x-slot>
        <div>
            <x-text-input wire:model="villa_name">
                <x-slot name="label">Primera Etiqueta</x-slot>
            </x-text-input>
        </div>
        <x-slot name="footer">
            <x-action-button>Guardar Información</x-action-button>
        </x-slot>
    </x-modal>
</div>
```

Any help would be super appreciated!!

**Comment by joshhanley** [comment, marked as answer]
@eddysantos when you are using `{{ $attributes }}` in your blade component it will also be outputting `wire:model=...`.

So if you have a look at the rendered html in your browser dev tools, you will see multiple `wire:model` being output where you didn't intend to.

Try changing it so you don't output the `wire:model` by using `{{ $attributes->whereDoesntStartWith('wire:model') }}`

Hope this helps!

**Reply by eddysantos** [reply]
Thanks for taking the time to answer! However, the point is to pass a "wire:model" attribute to the input element, so it can map to the Livewire component's property. Is this not the way to do it?

**Reply by joshhanley** [reply]
@eddysantos yes, generally, but not if you are using `@entangle` or if you have already output `wire:model` in your component, you don't want to output it twice.

So in both those scenarios you need to filter out the `wire:model` so it doesn't get output.

For example, with your `text-input.blade.php`

```blade
<div {{ $attributes->merge(['class'=> 'flex flex-col']) }}> {{-- `wire:model` is being output here --}}
    <label for="{{ $id }}">{{ $label }}</label>
    <input
        type="text"
        class="py-1 px-2"
        id="{{ $id }}"
        placeholder="{{ $placeholder }}"
        {{ $attributes->wire('model') }} {{-- `wire:model` is also being output here --}}
    >
</div>
```

You can see where I have put the comments, that `wire:model` is being output twice. So you need to change the first div to this, to stop it outputting `wire:model` as well.

```blade
<div {{ $attributes->whereDoesntStartWith('wire:model')->merge(['class'=> 'flex flex-col']) }}>
```

Hope this helps!

**Reply by eddysantos** [reply]
@joshhanley  Thank you very much! In the first comment I thought you meant not to include the wire:model in the input component, as it was already on the modal. I was oblivious to the fact that when I was calling $attributes->merge(....), not only I was merging the class attribute, but also duplicating the wire:model. 

It works now! Thanks so much.

**Reply by joshhanley** [reply]
@eddysantos no worries! 🙂
